### PR TITLE
Enums

### LINKED ISSUE #28: Convert enums
When enums are not the same class, it should try to convert the values by name. Aka, the target has to be a superset of the source. Using `ignore`, `from`, `custom`, and defaults should work the same way as for classes.

- [ ] enums
- [ ] enums with values

### PR DESCRIPTION
**Pull request by tenhobi**
Fixes #28

### CODE REVIEW

File: packages/auto_mappr/analysis_options.yaml [hunk 1]
```diff
@@ -5,6 +5,9 @@ analyzer:
   plugins:
     - dart_code_metrics
 
+  exclude:
+    - test/integration/fixture/error/**
```
Comment (tenhobi) [inline packages/auto_mappr/analysis_options.yaml:9]: I think having `test/integration/error_fixture` would be better.

File: packages/auto_mappr/lib/src/builder/map_model_body_method_builder.dart [hunk 2]
```diff
@@ -36,6 +37,17 @@ class MapModelBodyMethodBuilder {
     // Add handling of whenSourceIsNull.
     block.statements.add(_whenModelIsNullHandling());
 
+    // Is there an enum involved in the mapping?
+    if (mapping.isEnumMapping) {
+      final enumBuilder = EnumAssignmentBuilder(mapperConfig: mapperConfig, mapping: mapping);
+
+      final expression = enumBuilder.build();
+
+      block.statements.add(expression.statement);
```
Comment (tenhobi) [inline packages/auto_mappr/lib/src/builder/map_model_body_method_builder.dart:46]: Not sure if the concrete builder shouldn't have do the statement itself?

File: packages/auto_mappr/analysis_options.yaml (hunk 1 above)
Comment (petrnymsa) [reply packages/auto_mappr/analysis_options.yaml]: Agree

File: packages/auto_mappr/test/integration/fixture/error/enum_mapping_not_enum.dart [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import 'package:auto_mappr_annotation/auto_mappr_annotation.dart';
+
+part 'enum_mapping_subset.g.dart';
```
[the remark this answers is not in the document]
Comment (petrnymsa) [reply packages/auto_mappr/test/integration/fixture/error/enum_mapping_not_enum.dart]: Fixed

**Comment by tenhobi** [comment]
jiffy

**Review by petrnymsa — APPROVED** [review]

**Pull request merged**